src/modules/sbstudio/api/gateway.py

```diff
@@ -0,0 +1,105 @@
+from typing import Any
+
+from sbstudio.plugin.utils.progress import ProgressReport
+
+from .base import SkybrushStudioBaseAPI
+from .errors import SkybrushStudioAPIError
+
+
+__all__ = ("SkybrushGatewayAPI",)
+
+
+class SkybrushGatewayAPI(SkybrushStudioBaseAPI):
+    """Class that represents a connection to the API of a
+    Skybrush Gateway for request signing and progress display.
+    """
+
+    _progress_task_url: str | None = None
```

Comment:
This does not belong here as (in theory) you could be performing multiple progress-related operations in parallel and then you would need to maintain two such URLs.

A better pattern would be to use a context manager. I will elaborate on this idea in this PR later.